src/time/src/mcp_server_time/server.py

```diff
@@ -0,0 +1,145 @@
+from datetime import datetime
+import json
+from typing import Dict, Any, Optional, Sequence
+
+import pytz
+from tzlocal import get_localzone
+from mcp.server import Server
+from mcp.server.stdio import stdio_server
+from mcp.types import Tool, TextContent, ImageContent, EmbeddedResource
+
+
+class TimeServer:
+    def __init__(self, local_tz_override: Optional[str] = None):
+        self.local_tz = pytz.timezone(local_tz_override) if local_tz_override else get_localzone()
+
+    def get_current_time(self, timezone_name: str | None = None) -> Dict[str, Any]:
+        """Get current time in specified timezone or local timezone if none specified"""
+        timezone = pytz.timezone(timezone_name) if timezone_name else self.local_tz
+        current_time = datetime.now(timezone)
+        
+        return {
+            "timezone": timezone_name or str(self.local_tz),
+            "time": current_time.strftime("%H:%M %Z"),
+            "date": current_time.strftime("%Y-%m-%d"),
+            "full_datetime": current_time.strftime("%Y-%m-%d %H:%M:%S %Z"),
+            "is_dst": bool(current_time.dst())
+        }
+
+    def convert_time(self, source_tz: str, time_str: str, target_tz: str) -> Dict[str, Any]:
```

Comment:
Note taht we actually know the exact shape. We should return a dataclass, TypedDict or pydantic model.